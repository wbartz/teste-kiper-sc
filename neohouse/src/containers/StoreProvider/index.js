import PropTypes from 'prop-types';
import React, { createContext, useReducer } from 'react';
import appActions from '../../actions';
import { useActions, LOG } from '../../helpers';
import appReducers, { initialState as appInitialState } from '../../reducers';

const CreateProviderValue = (reducer, initialState, actions) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const boundActions = useActions(actions, dispatch);

  return { ...boundActions, ...state };
};

export const AppContext = createContext();

const StoreProvider = ({ children }) => {
  LOG('Context start');

  return (
    <AppContext.Provider
      value={CreateProviderValue(appReducers, appInitialState, appActions)}
    >
      {children}
    </AppContext.Provider>
  );
};

StoreProvider.propTypes = {
  children: PropTypes.node,
};

StoreProvider.defaultProps = {
  children: null,
};

export default StoreProvider;
