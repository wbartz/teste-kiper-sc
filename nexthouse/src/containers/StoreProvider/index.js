import PropTypes from "prop-types";
import React from "react";
import appActions from "../../actions";
import { useActions } from "../../helpers";
import appReducers, { initialState as appInitialState } from "../../reducers";

const CreateProviderValue = (reducer, initialState, actions) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const boundActions = useActions(actions, dispatch);

  return { ...boundActions, ...state };
};

export const AppContext = createContext();

const StoreProvider = ({ children }) => {
  return (
    <AppContext.Provider
      value={CreateProviderValue(appReducers, appInitialState, appActions)}
    >
      {children}
    </AppContext.Provider>
  );
};

StoreProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default StoreProvider;
