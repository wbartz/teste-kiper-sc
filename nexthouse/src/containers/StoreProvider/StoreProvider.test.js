import { shallow } from 'enzyme';
import React from 'react';
import StoreProvider from '.';

describe('StoreProvider', () => {
  it('render without crashing', () => {
    expect(shallow(<StoreProvider />)).toMatchSnapshot();
  });
});
