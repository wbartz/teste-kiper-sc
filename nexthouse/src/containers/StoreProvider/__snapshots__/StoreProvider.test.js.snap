// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreProvider render without crashing 1`] = `ShallowWrapper {}`;
